actionpack/lib/action_dispatch/system_testing/test_helpers/setup_and_teardown.rb

```diff
@@ -21,6 +21,12 @@ def after_teardown
           Capybara.reset_sessions!
           super
         end
+
+        private
+
+          def app_host
+            Capybara.app_host || DEFAULT_HOST
```

Comment:
This change break in the case someone changes the `app_host` outside a callback. Let's say capybara starts to set a value by default, Rails will not be able to do se the value anymore.